Classes/Mapper/Event.php

```diff
@@ -174,15 +214,39 @@ public function findForAutomaticStatusChange(): Collection
      */
     public function findForRegistrationDigestEmail(): Collection
     {
-        $whereClause = 'registrations <> 0' .
-            ' AND object_type <> ' . \Tx_Seminars_Model_Event::TYPE_TOPIC .
+        $whereClause = 'registrations <> 0 AND object_type <> ' . \Tx_Seminars_Model_Event::TYPE_TOPIC .
+            ' AND hidden = 0 AND deleted = 0 ' .
             ' AND EXISTS (' .
             'SELECT * FROM tx_seminars_attendances ' .
             'WHERE tx_seminars_attendances.deleted = 0 ' .
             ' AND tx_seminars_attendances.seminar = tx_seminars_seminars.uid' .
             ' AND tx_seminars_attendances.crdate > tx_seminars_seminars.date_of_last_registration_digest' .
             ')';
 
-        return $this->findByWhereClause($whereClause, 'begin_date ASC');
+        $sql = 'SELECT * FROM ' . $this->getTableName() . ' WHERE ' . $whereClause . ' ORDER BY begin_date ASC';
+
+        /** @var Connection $connection */
+        $connection = $this->getConnectionForTable($this->getTableName());
+        $statement = $connection->prepare($sql);
+        $statement->execute();
+        $rows = $statement->fetchAll();
+
+        return $this->getListOfModels($rows);
+    }
+
+    private function addSortingForBackend(QueryBuilder $queryBuilder)
```

Comment:
Now that I think of it, we actually do not need the TCA sorting for these three methods. To keep things reproducible, we can sort by UID in ascending order.